### PR TITLE
Support airplane dev for views

### PR DESCRIPTION
**Pull request by leeweisberger**
## Description
Airplane dev on a view directory spins up a local server and runs the view. We pass through authn, host, and env information from CLI and run the view as a vite app.

We're eventually going to make a few changes here, but this is a start:
- Allow this to work if we are nested inside of a view, not just at the top level
- Support the view definition not being in the same place as the view root
- (Maybe) support overriding the `dev` command with a custom script.

## Test plan
I ran `airplane dev` on one of the example views and it worked!

FIXES AIR-3958

### CODE REVIEW

**Review by rose-huang — CHANGES_REQUESTED** [review]
Just curious - can we make `airplane dev` work without having the `vite.config.ts` file? Following up on Josh's question about not exposing this config file https://airplane.slab.com/posts/2022-06-07-views-definition-umob2ewz#thread-0rl56gee 

For deploying, it seems like we could just generate the file with the config after the user runs `airplane deploy` but not sure if we would want to create a config file during `airplane dev` and remove it afterwards?

File: pkg/cmd/tasks/dev/dev.go [hunk 1]
```diff
@@ -34,10 +34,12 @@ import (
 )
 
 type config struct {
-	root    *cli.Config
-	file    string
-	args    []string
-	envSlug string
+	root      *cli.Config
+	fileOrDir string
```
[the remark this answers is not in the document]
Comment (leeweisberger) [reply pkg/cmd/tasks/dev/dev.go:40]: I updated the pr and created a new views folder / subcommands. It's a bit wonky where:
- views dev: behaves as expected
- views deploy: calls through to task deploy
- tasks dev: behaves as expected unless we are in a view and then we call through to the views dev.

**Review by rose-huang — APPROVED** [review]
> views dev: behaves as expected
views deploy: calls through to task deploy
tasks dev: behaves as expected unless we are in a view and then we call through to the views dev.

OK this seems ok for now but I think we can have a follow up discussion on how commands will work and make sure the descriptions are consistent. I think it would make sense that:

airplane dev - will dev a task or view depending on which is passed in
airplane deploy - will dev a task or view depending on which is passed in

airplane views dev/deploy - will only dev or deploy a view and if you give it a task, it will tell you to try airplane tasks dev/deploy
airplane tasks dev/deploy - will only dev or deploy a task and if you give it a view, it will tell you to try airplane views dev/deploy

Also not sure what the reason is for previously having both `airplane tasks dev` and `airplane dev` ... do we need both? 


Made a task: https://linear.app/airplane/issue/AIR-3983/figure-out-how-airplane-cli-devdeploy-commands-should-work-for-views

**Comment by leeweisberger** [comment]
> 

airplane dev is currently an alias (shortcut) of airplane tasks dev. 

Yeah I think we probably want to remove that alias and create a parent `airpalne dev` command that determines the type and then calls the correct command.

**Pull request merged**